src/System.IO.FileSystem/tests/Directory/GetDirectories_str.cs

```diff
@@ -1,377 +1,54 @@
-
-            strLoc = "Loc_fy87e";
-
-            dirArr = dir2.GetDirectories("Test1Dir2");
-            iCountTestcases++;
-            if (dirArr.Length != 1)
-            {
-                iCountErrors++;
-                printerr("Error_2398v! Incorrect number of directories returned, expected==1, got==" + dirArr.Length);
-            }
-            iCountTestcases++;
-            if (!dirArr[0].Name.Equals("Test1Dir2"))
-            {
-                iCountErrors++;
-                printerr("Error_88gbb! Incorrect directory returned==" + dirArr[0]);
-            }
-
-            // [] Multiple wildcards
-
-            strLoc = "Loc_48yg7";
-
-            dirArr = dir2.GetDirectories("T*st*D*2");
-            iCountTestcases++;
-            if (dirArr.Length != 2)
-            {
-                iCountErrors++;
-                printerr("Error_39g8b! Incorrect number of directories returned, expected==2, got==" + dirArr.Length);
-            }
-
-
-            // [] Searchstring starting and ending with '*'
-            // [] Search is case insensitive
-            dir2.CreateSubdirectory("AAABB");
-            dir2.CreateSubdirectory("aaabbcc");
+            return Directory.GetDirectories(path);
+        }
 
-            dirArr = dir2.GetDirectories("*BB*");
-            iCountTestcases++;
-            if (dirArr.Length != (Interop.IsWindows ? 2 : 1))
-            {
-                iCountErrors++;
-                printerr("Error_4y190! Incorrect number of directories==" + dirArr.Length);
-            }
-            names = new String[2];
-            i = 0;
-            foreach (DirectoryInfo d in dirArr)
-                names[i++] = d.Name;
+        #endregion
 
-            iCountTestcases++;
-            if (Array.IndexOf(names, "AAABB") < 0)
-            {
-                iCountErrors++;
-                printerr("Error_956yb! Incorrect name==" + dirArr[0]);
-            }
+        #region PlatformSpecific
 
-            if (Interop.IsWindows)
+        [Fact]
+        [PlatformSpecific(PlatformID.AnyUnix)]
+        public void EnumerateWithSymLinkToDirectory()
+        {
+            using (var containingFolder = new TemporaryDirectory())
             {
-                iCountTestcases++;
-                if (Array.IndexOf(names, "aaabbcc") < 0)
+                // Test a symlink to a directory that does and then doesn't exist
+                using (var targetDir = new TemporaryDirectory())
                 {
-                    iCountErrors++;
-                    printerr("Error_48yg7! Incorrect name==" + dirArr[1]);
+                    // Create a symlink to a folder that exists
+                    string linkPath = Path.Combine(containingFolder.Path, Path.GetRandomFileName());
+                    Assert.Equal(0, symlink(targetDir.Path, linkPath));
+                    Assert.True(Directory.Exists(linkPath));
+                    Assert.Equal(1, Directory.GetDirectories(containingFolder.Path).Count());
```

Comment:
These Directory.* calls should now be using the instance Get* methods on this test class, right?